### PR TITLE
Fixes for RestConnectionHandler, to work as expected against normal clients.

### PR DESCRIPTION
**Pull request by eriksunden**
Fixes in the RestConnectionHandler, to uphold the standards that:

1. The client is the one that closes the connection.
2. All request (both GET and POST) should return a message as well as status code.

Also did implement the possibility of having parameters for a GET, which is parsed correctly.
Though, the start/stop methods are only POST for now, as that is best practice.

### CODE REVIEW

File: src/ctroll/restconnectionhandler.cpp [hunk 1]
```diff
@@ -70,9 +70,45 @@ namespace {
...
+    void Debug(std::string msg) {
+        ::Debug("RestConnectionHandler", std::move(msg));
+    }
+
+    void Log(std::string msg) {
+        ::Log("REST", std::move(msg));
+    }
+
+    void sendResponse(QTcpSocket& socket, Response response, std::string content)
     {
```
Comment (alexanderbock) [inline src/ctroll/restconnectionhandler.cpp:82]:
```suggestion
    void sendResponse(QTcpSocket& socket, Response response, std::string content) {
```

File: src/ctroll/restconnectionhandler.cpp [hunk 2]
```diff
@@ -70,9 +70,45 @@ namespace {
...
+        if (content.empty())
+            return;
```
Comment (alexanderbock) [inline src/ctroll/restconnectionhandler.cpp:84]:
```suggestion
        if (content.empty()) {
            return;
        }
```

File: src/ctroll/restconnectionhandler.cpp [hunk 3]
```diff
@@ -70,9 +70,45 @@ namespace {
...
+
+        Debug(content);
+
+        std::string_view code = [](Response resp) {
+            switch (resp) {
+            case Response::Ok: return "200 OK";
+            case Response::BadRequest: return "400 Bad Request";
+            case Response::Unauthorized: return "401 Unauthorized";
+            case Response::Forbidden: return "403 Forbidden";
+            case Response::NotFound: return "404 Not Found";
```
Comment (alexanderbock) [inline src/ctroll/restconnectionhandler.cpp:94]:
```suggestion
                case Response::Ok: return "200 OK";
                case Response::BadRequest: return "400 Bad Request";
                case Response::Unauthorized: return "401 Unauthorized";
                case Response::Forbidden: return "403 Forbidden";
                case Response::NotFound: return "404 Not Found";
```

File: src/ctroll/restconnectionhandler.cpp [hunk 4]
```diff
@@ -248,7 +270,7 @@ void RestConnectionHandler::newConnectionEstablished() {
             "New connection from {}", socket->peerAddress().toString().toStdString()
         ));
 
-        connect(socket, &QTcpSocket::disconnected, socket, &QTcpSocket::deleteLater);
+        //connect(socket, &QTcpSocket::disconnected, socket, &QTcpSocket::deleteLater);
```
Comment (alexanderbock) [inline src/ctroll/restconnectionhandler.cpp:273]: If we don't need the line, we can remove it entirely

File: src/ctroll/restconnectionhandler.cpp [hunk 5]
```diff
@@ -284,53 +305,63 @@ void RestConnectionHandler::handleNewConnection() {
...
+    if (method == HttpMethod::Get && endpointValue.has_value()) {
+        QString endPointStr = QString::fromStdString(*endpointValue);
+        QStringList endPointList = endPointStr.split('?');
+        foundEndPoint = parseEndpoint(endPointList[0].toStdString());
+        if (endPointList.size() > 1) {
+            endPointList.pop_front();
+            params = parameters(endPointList);
+        }
+    }
+    else if(method == HttpMethod::Post) {
```
Comment (alexanderbock) [inline src/ctroll/restconnectionhandler.cpp:330]:
```suggestion
    else if (method == HttpMethod::Post) {
```

**Review by alexanderbock — APPROVED** [review]

**Review by eriksunden — COMMENTED** [review]
Solid

**Pull request merged**